cmd/edgectl/aes_install.go

```diff
@@ -527,149 +548,130 @@ func (i *Installer) Perform(kcontext string) error {
-		// verify the existence of an Ambassador deployment in the Ambassador
-		// namespace.
-		installedVersion, err := i.GetInstalledImageVersion()
+		i.show.Println("Failed to look for an existing installation:", err)
+		installedVersion = "" // Things will likely fail when we try to apply manifests
+	}
+
+	alreadyInstalled := false
+	if installedVersion != "" {
+		alreadyInstalled = true
+	} else {
 		if err != nil {
-			i.show.Println("Failed to look for an existing installation:", err)
-			installedVersion = ""
-			// Things will likely fail when we try to apply manifests
+			i.log.Printf("when obtaining the chart manager: %s", err)
+			i.Report("fail_download", ScoutMeta{"err", err.Error()})
+			return err
 		}
-		switch {
-		case i.version == installedVersion:
-			alreadyApplied = true
-			i.ShowWrapped(fmt.Sprintf("-> Found an existing installation of Ambassador Edge Stack %s.", i.version))
-		case installedVersion != "":
-			i.ShowWrapped(fmt.Sprintf("-> Found an existing installation of Ambassador Edge Stack %s.", installedVersion))
-			i.show.Println()
-			i.ShowWrapped(abortExisting)
-			i.show.Println()
-			i.ShowWrapped(seeDocs)
-			i.Report("fail_existing_aes", ScoutMeta{"installing", i.version}, ScoutMeta{"found", installedVersion})
-			return errors.Errorf("existing AES %s found when installing AES %s", installedVersion, i.version)
-		default:
-			i.ShowWrapped(abortCRDs)
-			i.show.Println()
-			i.ShowWrapped(seeDocs)
-			i.Report("fail_existing_crds")
-			return errors.New("CRDs found")
+
+		if err := chartRel.Sync(i.ctx); err != nil {
+			i.log.Printf("Failed to sync release: %s", err)
+			i.Report("fail_download", ScoutMeta{"err", err.Error()})
+			return err
 		}
-	}
 
-	if !alreadyApplied {
-		// Install the AES manifests
+		if chartRel.IsInstalled() {
+			// There is an Ambassador Helm chart installed. Don't do anything.
+			alreadyInstalled = true
+			i.SetMetadatum("Cluster Info", "managed", "helm")
+		}
+	}
 
-		i.ShowWrapped("-> Downloading images. (This may take a minute.)")
+	if alreadyInstalled {
+		i.ShowWrapped(fmt.Sprintf("-> Found an existing installation of Ambassador Edge Stack %s.", installedVersion))
+		i.Report("deploy", ScoutMeta{"already_installed", true})
 
-		if err := i.ShowKubectl("install CRDs", crdManifests, "apply", "-f", "-"); err != nil {
-			i.Report("fail_install_crds")
-			return err
+		switch installedFamily {
+		case "oss", "aes":
+			i.ShowWrapped(abortExisting, seeDocs)
+			i.Report("fail_existing_aes", ScoutMeta{"installing", i.version}, ScoutMeta{"found", installedVersion})
+			return errors.Errorf("existing AES %s found when installing AES %s", installedVersion, i.version)
+		default:
+			// continue with the rest of the setup
 		}
+	} else {
+		// Ambassador is definetively not installed: perform the installation
+		i.SetMetadatum("AES version being installed", "aes_version", i.version)
+		i.SetMetadatum("Cluster Info", "managed", "helm")
```

Comment:
If we don't have a previous installation, the `managed` metadata should not be set.